src/suggestion-box.ts

```diff
@@ -0,0 +1,409 @@
+                );
+                selectedItem.classList.add("sgn-active");
+            }
+        } else if (this.index == resultLength && this.has_more) {
+            this.private_items[this.index].classList.add("sgn-active");
+            this.private_items[this.prevIndex]?.classList.remove("sgn-active");
+        }
+
+        this.prevIndex = this.index;
+    }
+
+    private _setInputValue() {
+        const selectedValue = this.result[this.index][this.listkey] || this.result[this.index].name;
+        if (selectedValue) {
+            (this.input_el as HTMLInputElement).value = selectedValue;
+            this.show_container = false;
+        }
+    }
+    _highlight(name: any, input: string) {
+        if (input && name.toLowerCase().includes(input)) {
+            const parts = name.split(new RegExp(`(${input})`, "gi"));
+            return parts
+                .map((part: any, index: number) => {
+                    if (index > 1) {
+                        return `<span class="sgn-highlight">${part}</span>`;
+                    } else {
+                        return part;
+                    }
+                })
+                .join("");
+        }
+        return name;
+    }
+    _startsWithHighlight(name: any) {
+        const input_val = (this.input_el as HTMLInputElement)?.value;
+        if (!input_val) {
+            return name;
+        }
+
+        const escaped_input_val = input_val.replace(
+            /[-\/\\^$*+?.()|[\]{}]/g,
+            "\\$&"
+        );
+        const re = new RegExp(
+            "(" + escaped_input_val.split(" ").join("|") + ")",
+            "gi"
+        );
+        return name.replace(re, `<span class="sgn-highlight">$1</span>`);
+    }
+    setSuggestion(
+        suggestions: Array<{ [key: string]: string }>,
+        has_more: boolean,
+        key: string
+    ) {
+        this.listkey = key;
+        if ((this.input_el as HTMLInputElement)?.value.length >= this.minchar) {
+            this.result = [...suggestions];
+            this.index = -1;
+            let input_temp: HTMLElement | null | string = null;
+            input_temp = (this.input_el as HTMLInputElement)?.value;
+            input_temp ? (this.show_empty = false) : (this.show_empty = true);
+            this.has_more = has_more;
+            if (this.private_items?.length) {
+                for (let el of this.private_items) el.remove();
+            }
+            this.size = suggestions.length;
+            this.filtered_list = suggestions;
+            this.show_container = true;
+            for (let i of suggestions) {
+                let el = document.createElement("div");
+                el.className = "sgn-item";
+                el.slot = "__private-item";
+                console.log("key value ", i);
+                el.innerHTML =
+                    i.html ||
+                    (this.template_slots[0] == null || undefined
+                        ? this.highlight == "matched"
```

Comment:
Same as a previous comment. What if the highlight is null? I think you can move this logic to a private function where you can have certain checks and call that function in both the places.